rero/rero-ils
# user: sign in with a username

* Adds a username to all fixtures.
* Adds a new invenio accounts login view REST API.
* Adds a js code to the login view to be able to be logged using the
  username or the email.
* Renames the login (/signin) and logout (/signout) url to avoid
  conflict with the accounts REST API.
* Changes the link between the patron and invenio user link from email
  to id.
* Synchronize the data between the patron and user profile in both
  direction.
* Sets the default user password as the birth date.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/1590

## How to test?

- Bootstrap/Setup
- Login with a librarian and create a new patron.
- The new patron can be logged using a username.
- Change the data in the patron and professional view the data should be modified in both direction.

## Code review check list
